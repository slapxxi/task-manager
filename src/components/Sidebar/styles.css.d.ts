export const container: string;
export const title: string;
export const list: string;
export const listItem: string;
export const starIcon: string;
export const link: string;
export const slot: string;
export const section: string;
export const sectionHeader: string;
export const sectionListItem: string;
export const button: string;
export const row: string;
export const footer: string;
