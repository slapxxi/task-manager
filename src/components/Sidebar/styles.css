.container {
  --primary-color: var(--middle-grey);
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  padding: 20px;
  height: 100%;
  background-color: var(--bg);
  max-width: 250px;
  width: 33%;
}

.container nav {
  width: 100%;
}

.title {
  padding: 0;
  margin: 0;
  font-size: 18px;
  font-family: 'Source Sans Pro';
  text-transform: uppercase;
  font-weight: bold;
  color: var(--light-grey);
}

.list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.listItem {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

.starIcon {
  --primary-color: transparent;
  --secondary-color: var(--middle-grey);
}

.listItem > svg {
  margin-right: 10px;
}

.link {
  text-decoration: none;
  color: var(--middle-grey);
  flex: 1;
}

.link:global(.active) {
  font-weight: bold;
  color: var(--dark-grey);
}

.slot {
  color: var(--middle-grey);
}

.section {
  margin: 20px 0;
}

.sectionHeader {
  --primary-color: var(--light-grey);
  display: flex;
  align-items: center;
  margin-bottom: 20px;
}

.sectionHeader > .title {
  flex: 1;
}

.sectionListItem {
  composes: listItem;
  margin-left: 10px;
  margin-bottom: 10px;
}

.button {
  --primary-color: var(--light-grey);
  font-size: 14px;
  color: var(--light-grey);
}

.button:hover {
  --primary-color: var(--dark-grey);
  color: var(--dark-grey);
}

.row {
  display: flex;
  align-items: center;
}

.footer {
  display: flex;
  align-items: center;
  margin-top: auto;
  justify-content: space-between;
}
