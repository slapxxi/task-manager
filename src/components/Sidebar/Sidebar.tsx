import { createProject, getProjectProgress } from '@lib';
import selectTodayTasks from '@lib/selectTodayTasks';
import Button from '@local/components/Button/Button';
import CalendarIcon from '@local/components/CalendarIcon/CalendarIcon';
import IconSystem from '@local/components/IconSystem/IconSystem';
import ProgressCheckbox from '@local/components/ProgressCheckbox/ProgressCheckbox';
import { useStore } from '@local/hooks';
import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import styles from './styles.css';

function Sidebar() {
  const { getProjects, getTasks, actions } = useStore();
  const todayTasks = selectTodayTasks(getTasks());

  function handleAddProject() {
    actions.updateProject(createProject({ name: 'Untitled' }));
  }

  return (
    <aside className={styles.container}>
      <header className={styles.header}>
        <IconSystem name="logo" size={20} /> <IconSystem name="dots" size={20} />{' '}
      </header>

      <nav>
        <ul className={styles.list}>
          <li className={styles.listItem}>
            <IconSystem name="inbox" size={20} />{' '}
            <NavLink to="/" className={styles.link} exact>
              Inbox
            </NavLink>{' '}
            <span className={styles.slot}>{getTasks().length}</span>
          </li>
          <li className={styles.listItem}>
            <IconSystem name="star" size={20} className={styles.starIcon} />{' '}
            <NavLink activeClassName="active" to="/today" exact className={styles.link}>
              Today
            </NavLink>
            <span className={styles.slot}>{todayTasks.length}</span>
          </li>
          <li className={styles.listItem}>
            <CalendarIcon date={new Date()} size={18} />
            <NavLink
              activeClassName="active"
              to="/upcoming"
              exact
              className={styles.link}
            >
              Schedule
            </NavLink>
          </li>
        </ul>
      </nav>

      <nav className={styles.section}>
        <header className={styles.sectionHeader}>
          <h1 className={styles.title}>
            <NavLink to="/projects" className={styles.link} exact>
              Projects
            </NavLink>
          </h1>
          <Button onClick={handleAddProject}>
            <IconSystem name="plus" size={12} />
          </Button>
        </header>

        <ul className={styles.list}>
          {getProjects().map((p) => (
            <li key={p.id} className={styles.sectionListItem}>
              <NavLink to={`/projects/${p.id}`} className={styles.link}>
                {p.name}
              </NavLink>
              <span className={styles.slot}>
                {getProjectProgress(p) === 1 ? (
                  <IconSystem name="checkmark" size={14} />
                ) : (
                  <ProgressCheckbox
                    progress={getProjectProgress(p)}
                    size={14}
                    animate={false}
                  />
                )}
              </span>
            </li>
          ))}
        </ul>
      </nav>

      <nav className={styles.section}>
        <Button className={styles.button}>
          <IconSystem name="plus" size={14} style={{ marginRight: 10 }} /> Add Workspace
        </Button>
      </nav>

      <footer className={styles.footer}>
        <Button className={styles.button}>
          <IconSystem name="plus" size={14} />
        </Button>
        <Link to="/settings" className={styles.button}>
          <IconSystem name="cog" size={16} />
        </Link>
      </footer>
    </aside>
  );
}

export default Sidebar;
