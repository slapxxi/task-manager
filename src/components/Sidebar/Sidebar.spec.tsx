import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';
import Sidebar from './Sidebar';

jest.mock('@local/components/IconSystem/IconSystem');
jest.mock('@local/components/CalendarIcon/CalendarIcon');

it('renders', () => {
  const { container } = render(
    <Router>
      <Sidebar />
    </Router>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
