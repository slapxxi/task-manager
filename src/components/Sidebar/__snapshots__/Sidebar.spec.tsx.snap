// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<aside
  class="container"
>
  <header
    class="header"
  >
    <div
      name="logo"
      size="20"
    >
      icon-
      logo
    </div>
     
    <div
      name="dots"
      size="20"
    >
      icon-
      dots
    </div>
     
  </header>
  <nav>
    <ul
      class="list"
    >
      <li
        class="listItem"
      >
        <div
          name="inbox"
          size="20"
        >
          icon-
          inbox
        </div>
         
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          Inbox
        </a>
         
        <span
          class="slot"
        >
          0
        </span>
      </li>
      <li
        class="listItem"
      >
        <div
          class="starIcon"
          name="star"
          size="20"
        >
          icon-
          star
        </div>
         
        <a
          class="link"
          href="/today"
        >
          Today
        </a>
        <span
          class="slot"
        >
          0
        </span>
      </li>
      <li
        class="listItem"
      >
        <div
          class="calendar-icon"
        >
          calendar-icon
        </div>
        <a
          class="link"
          href="/upcoming"
        >
          Schedule
        </a>
      </li>
    </ul>
  </nav>
  <nav
    class="section"
  >
    <header
      class="sectionHeader"
    >
      <h1
        class="title"
      >
        <a
          class="link"
          href="/projects"
        >
          Projects
        </a>
      </h1>
      <button
        class="sc-bdVaJa fFWIpk"
      >
        <div
          name="plus"
          size="12"
        >
          icon-
          plus
        </div>
      </button>
    </header>
    <ul
      class="list"
    />
  </nav>
  <nav
    class="section"
  >
    <button
      class="button sc-bdVaJa fFWIpk"
    >
      <div
        name="plus"
        size="14"
        style="margin-right: 10px;"
      >
        icon-
        plus
      </div>
       Add Workspace
    </button>
  </nav>
  <footer
    class="footer"
  >
    <button
      class="button sc-bdVaJa fFWIpk"
    >
      <div
        name="plus"
        size="14"
      >
        icon-
        plus
      </div>
    </button>
    <a
      class="button"
      href="/settings"
    >
      <div
        name="cog"
        size="16"
      >
        icon-
        cog
      </div>
    </a>
  </footer>
</aside>
`;
